core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -1691,7 +1695,7 @@ public static boolean isValidSortOption(String sortScope) {
     if (sortScope == null) {
       return false;
     }
-    switch (sortScope.toUpperCase()) {
+    switch (sortScope.toUpperCase(Locale.ENGLISH)) {
```

Comment:
Should not use Locale.ENGLISH.Refer to String documentation in idk
use Locale.ROOT